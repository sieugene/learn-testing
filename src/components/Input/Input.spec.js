import React from 'react'
import Input from './Input'
import { shallow } from 'enzyme';

describe(("Input component"), () => {
    it("should test", () => {
        const component = shallow(<Input/>)
        expect(component).toMatchSnapshot();
    })
    describe("default props", () => {
        it("should use default props onChange", () => {
            const result = Input.defaultProps.onChange();
            expect(result).toBe(undefined)
        })
        it("should use default props onKeyPress", () => {
            const result = Input.defaultProps.onKeyPress();
            expect(result).toBe(undefined)
        })
        it("should use default props value", () => {
            const result = Input.defaultProps.value;
            expect(result).toBe("")
        })
    })
})
