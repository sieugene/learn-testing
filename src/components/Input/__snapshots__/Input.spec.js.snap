// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should test 1`] = `
<div
  className="inputWrapper"
>
  <i
    className="fas fa-search"
  />
  <input
    className="input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Click to search"
    value=""
  />
</div>
`;
